crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -851,6 +852,50 @@ impl EntityCommands<'_> {
         self.add(try_insert(bundle))
     }
 
+    /// Adds a dynamic component to the entity.
+    ///
+    /// This will overwrite any previous value(s) of the same component type.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// // Note: this is a fairly contrived example, mostly because most uses of dynamic components are
+    /// // complex and abstract
```

Comment:
Personally I would avoid this note. Maybe replace it with a link to some central place for documentation on dynamic components that also shows some concrete use cases, so people can have an idea of what they can be used for.